Description: Change d_fmt delimiter from - to /
Bug-Ubuntu: https://launchpad.net/bugs/1101134
Forwarded: https://sourceware.org/bugzilla/show_bug.cgi?id=17438
Author: Gunnar Hjalmarsson <gunnarhj@ubuntu.com>

--- a/localedata/locales/pt_BR	2012-03-06 10:10:19.000000000 +0100
+++ b/localedata/locales/pt_BR	2014-09-27 09:30:29.863838693 +0200
@@ -113,7 +113,7 @@
 	"<U006E><U006F><U0076><U0065><U006D><U0062><U0072><U006F>";/
 	"<U0064><U0065><U007A><U0065><U006D><U0062><U0072><U006F>"
 d_t_fmt "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
-d_fmt   "<U0025><U0064><U002D><U0025><U006D><U002D><U0025><U0059>"
+d_fmt   "<U0025><U0064><U002F><U0025><U006D><U002F><U0025><U0059>"
 t_fmt   "<U0025><U0054>"
 am_pm   "";""
 t_fmt_ampm ""
