--- glibc/elf/dl-load.c	2013-01-23 20:55:08.648614327 -0700
+++ glibc/elf/dl-load.c	2013-01-23 20:55:52.852612764 -0700
@@ -2008,6 +2008,12 @@
 	      goto close_and_out;
 	    }
 #endif
+	  else if (! __builtin_expect (elf_machine_matches_host (ehdr), 1))
+	    {
+	      /* Another non-fatal error, let's skip right past the
+	         the libraries obviously built for other machines.  */
+	      goto close_and_out;
+	    }
 	  else if (ehdr->e_ident[EI_DATA] != byteorder)
 	    {
 	      if (BYTE_ORDER == BIG_ENDIAN)
